### ISSUE TITLE
LF test, with Noise

### ISSUE THREAD

**Issue by evanbiederstedt** [evanbiederstedt/CMBintheLikeHoodz#4]
This issue refers to notebook: "July1_CMB_Likelihoods_LFtestC3_withNoise.ipynb"

https://github.com/evanbiederstedt/CMBintheLikeHoodz/blob/master/Likelihood_Comparison/July1_CMB_Likelihoods_LFtestC3_withNoise.ipynb

The results for the LF are still strange. Parameters initialized to expected C_3 value and noise sigma^2 initialized to empirical variance, Var(a3m). 

sigma2 = np.linspace(5e-12, 5e-15, num=20 )          #set default num = 20
vary_x_samples1 = np.linspace(1e-10, 9e-10, num=20 ) #set default num = 20

LogLike_wNoise(vary_x_samples1, sigma2)
Out[93]:
array([  7.95859016e+05,   8.43221710e+05,   8.96161152e+05,
         9.55721670e+05,   1.02322633e+06,   1.10037596e+06,
         1.18939415e+06,   1.29324475e+06,   1.41596776e+06,
         1.56321647e+06,   1.74315196e+06,   1.96800699e+06,
         2.25698671e+06,   2.64206127e+06,   3.18068375e+06,
         3.98749581e+06,   5.32904915e+06,   8.00003064e+06,
         1.59148294e+07,   8.56343453e+08])

There is a bug!

Possible reasons: 

I worry about the np.linalg.inv values for "invCij". They are huge!

**Comment by davidwhogg** [comment]
I have no idea why you are doing this test:

(a) fix data

(b) vary only one parameter at a time, don't vary the parameter and the data

(c) make grids in the log of parameters not linear

**Comment by evanbiederstedt** [comment]
RE: (a) & (b)

Yes, it makes no sense to vary both the data and the parameter (i.e. here, the C_3 and sigma^2). 

The correct thing to do it check each the 20 sigma^2 values each each C_3 LF. 

However, I'm only interested in the magnitudes of the LF using noise at the moment. 

RE:  "(c) make grids in the log of parameters not linear"

I am unsure what you mean. Please explain. 

At the moment, I simply generated 20 "C_3 values" from 1e-10 to 9e-10, and 20 "sigma^2" values around e-12. I then check what the LF outputs.

**Comment by davidwhogg** [comment]
I am not sure how you get that C_3 and sigma^2 is the "data and the parameter".  There are two parameters, there is one data set.  The two parameters you can vary together or separately as you like.  IT will be clearer to vary them separately.

Not sure what you don't understand about my suggestion to make the grids in the log of the parameter?  If you are varying a parameter Q over three orders of magnitude, you should make a linspace for lnQ not Q.

**Comment by evanbiederstedt** [comment]
There are two parameters, C_3 and noise sigma^2. 

In notebook "July1_CMB_Likelihoods_LFtestC3_withNoise.ipynb", where am I varying the data? 

What are you referring to when you state "don't vary the data"?

**Comment by davidwhogg** [comment]
maybe you have just named things strangely, but what is `vary_x_test` if not the data?  If it is C_3, perhaps you should call it `C_3`.  In general, the code should use the same names as the latex document as our discussion.

**Comment by evanbiederstedt** [comment]
I named things strangely. 

At some point in our e-mail correspondence, I began referring to "C_3" as "x", as in, 

"For our matrix S_ij = 7/4pi * x * P_3 (dotproduct_matrix), we vary this variable x". 

I am naming "C_3" as the value generated by the Boltzmann code for the theoretical value of C_3. 

I am doing this so as not to confuse myself between the parameter we are varying, initialized around values we expect (i.e. theoretical output CAMB C_3 and the empirical output from the map, \hat{C}_3) and the actual values (i.e. what I call in the code "C3" and "hatC3")
